src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -70,10 +75,55 @@ public String greeting(@RequestParam(name = "name", required = false, defaultVal
         return bookRepository.findByTitle(bookTitle);
     }
 
-    @GetMapping("/publisherGenreYear/{publisher}/{genre}/{year}")
-    public List<Book> findByGenrePublisherYear(@PathVariable String publisher, @PathVariable String genre,
-            @PathVariable String year) {
-        return bookRepository.findByPublisherAndGenreAndYear(publisher, genre, year);
+    @GetMapping("/publisherGenreYear")
+    public Page<Book> findByGenrePublisherYear(@RequestParam(required = false) String publisher,
+            @RequestParam(required = false) String genre,
+            @RequestParam(required = false) String year,
+            @PageableDefault(sort = {Constants.SORT_BOOK}, value = 5) Pageable pageable) {
+
+        return bookRepository.findByPublisherAndGenreAndYear(publisher, genre, year, pageable);
+    }
+
+    @GetMapping("/allBooks")
```

Comment:
You can replace @GetMapping("/allBooks") to @GetMapping("/"), finally this endpoint returns all values if all params are null

Reply:
Fixed at last commit!